File: packages/ts/staffkm-sdk/src/resources/workspaces.ts
import { AxiosInstance } from 'axios'

export class WorkspacesResource {
  constructor(private http: AxiosInstance) {}

  async list(): Promise<any[]> {
    const r = await this.http.get('/api/v1/workspaces')
    return r.data?.data ?? []
  }

  async create(name: string, slug?: string): Promise<any> {
    const r = await this.http.post('/api/v1/workspaces', { name, slug })
    return r.data?.data ?? r.data
  }

  async get(id: string): Promise<any> {
    const r = await this.http.get(`/api/v1/workspaces/${id}`)
    return r.data?.data ?? r.data
  }
}

File: packages/python/staffkm-sdk/staffkm/resources/workspaces.py
"""Workspaces resource."""
from __future__ import annotations

import httpx


class WorkspacesResource:
    def __init__(self, http: httpx.Client):
        self._http = http

    def list(self) -> list[dict]:
        r = self._http.get("/api/v1/workspaces")
        r.raise_for_status()
        return r.json().get("data") or []

    def create(self, name: str, slug: str | None = None) -> dict:
        body = {"name": name}
        if slug:
            body["slug"] = slug
        r = self._http.post("/api/v1/workspaces", json=body)
        r.raise_for_status()
        return r.json().get("data") or r.json()

    def get(self, workspace_id: str) -> dict:
        r = self._http.get(f"/api/v1/workspaces/{workspace_id}")
        r.raise_for_status()
        return r.json().get("data") or r.json()
